src/main/java/com/_up/megastore/services/implementations/CategoryService.java

```diff
@@ -41,28 +41,25 @@ private Category getSuperCategory(UUID superCategoryId) {
     }
 
     @Override
-    public CategoryResponse deleteCategory(UUID categoryId){
-        ifCategoryExistThrowException(categoryId);
+    public void deleteCategory(UUID categoryId){
+        Category category = findCategoryByIdOrThrowException(categoryId);
         ifCategorySubcategoriesExistThrowException(categoryId);
-        Category category = ifCategoryIsNotDeletedThrowException(categoryId);
+        getCategoryByIdAndDeletedFalseOrThrowException(category);
         category.setDeleted(true);
-        return CategoryMapper.toCategoryResponse(categoryRepository.save(category));
+        categoryRepository.save(category);
     }
 
     public void ifCategorySubcategoriesExistThrowException(UUID categoryId){
         if(!findCategoryByIdOrThrowException(categoryId).getSubCategories().isEmpty()){
             throw new DataIntegrityViolationException("Category with id "+ categoryId + " has subcategories and cannot be deleted.");
         }
     }
-    public void ifCategoryExistThrowException (UUID categoryId){
-        if (!categoryRepository.existsById(categoryId)) {
-            throw new NoSuchElementException("Category with id " + categoryId + " does not exist.");
+
+    public void getCategoryByIdAndDeletedFalseOrThrowException(Category category){
```

Comment:
Por qué este método se llama "getCategory...()" si no retorna ninguna categoria? Los nombres de los métodos deben ser representativos a lo que hacen, esto sería más algo como

```suggestion
    public void validateIfCategoryIsDeletedAndThrowException(Category category){
```